### ISSUE TITLE
Returning HTTP 503 when accessing via HAProxy

### ISSUE THREAD

**Issue by zenodotus280** [navidrome/navidrome#2034]
### Description

Navidrome returns HTTP 503 when accessed from HAProxy

### Expected Behaviour

I should get the regular login screen.

### Steps to reproduce

1. Start Navidrome using Docker or the standard Linux install instructions.
2. Verify that the server is available via http://LOCAL_IP_ADDRESS:4533
3. Configure based backend in HAProxy with no SSL options.
4. Add frontend to redirect based on ACL.

### Additional information

The backend and frontend are configured correctly: I swapped the backend from a known-good service and started to get 503.

I can access all other services externally using HAProxy (subdomain.example.com) as well as locally with IP or hostname.home.arpa; Navidrome is the only one that is unable to respond to navidrome.example.com

Previously successful using Nginx Proxy Manager.

**Comment by zenodotus280** [comment]
My backend named "navidrome"
```
| Mode | Name        | Forwardto     | Address       | Port | Encrypt(SSL) | SSL checks | Weight | Actions

| active | navidrome | Address+Port: | 192.168.1.149 | 4533 | no           | no         |  
```
</div></div></div></div>

**Comment by deluan** [comment]
Can you please provide the logs from the Navidrome server?

**Comment by TheWrathKing** [comment]
I am also running Navidrome behind HaProxy and it's working like a charm.
My configuration is quite simple for this one , in my frontend section I have something like this :
        acl navidrome-acl hdr(host) -i subdomain.domain.com
        use_backend navidrome if navidrome-acl
and my navidrome backend has only the following directive :
backend navidrome
        server raspberry-pi 192.168.X.X:4533
Also, my BaseUrl is left blank in the Navidrome configuration file (default) and I did not use the ReverseProxy... options

**Comment by zenodotus280** [comment]
Took me a while to figure out how/where the logging is done! I've set my logging to trace and monitored the output of `journalctl -fu navidrome` and I saw HTTP 200 responses for my IP and local domain name (navidrome.home.arpa) but not a single response when I try via navidrome.example.com

I'm suspecting it is _still_ an HAProxy issue despite setting up my front/backends as @TheWrathKing did. I tried to find others with the same problem but only discovered [someone who gave up](https://www.reddit.com/r/navidrome/comments/svdlbs/comment/hxfxb4g/?utm_source=share&utm_medium=web2x&context=3).

**Comment by zenodotus280** [comment]
Post-Mortem:
I'm running PFSense with HAProxy and the default "health check" for a given backend is to use some sort of HTTP request. I recall early in my troubleshooting seeing this in the HAProxy logs as a HTTP 405 (Method Not Allowed) but I didn't clue in to it. Thanks to [this tip-off](https://serverfault.com/questions/559624/haproxy-503-service-unavailable-no-server-is-available-to-handle-this-request/559625#559625) I can now move on with my life! 

Jellyfin and Audiobookshelf appear to respond to these health checks (whatever they might be) but Navidrome does not permit it.
